core/src/main/java/org/infinispan/persistence/manager/PersistenceManagerImpl.java

```diff
@@ -106,6 +105,7 @@ public void inject(AdvancedCache<Object, Object> cache, @ComponentName(CACHE_MAR
       this.persistenceExecutor = persistenceExecutor;
       this.byteBufferFactory = byteBufferFactory;
       this.marshalledEntryFactory = marshalledEntryFactory;
+      this.cacheStoreFactoryRegistry = cache.getComponentRegistry().getGlobalComponentRegistry().getComponent(CacheStoreFactoryRegistry.class);
```

Comment:
Yes, I've tried this approach but it resulted with the exception:
```
org.infinispan.commons.CacheConfigurationException: No registered default factory for component 'class org.infinispan.persistence.factory.CacheStoreFactoryRegistry' found!. To get more detail set the system property infinispan.debugDependencies to true
	at org.infinispan.factories.AbstractComponentRegistry.throwStackAwareConfigurationException(AbstractComponentRegistry.java:884)
	at org.infinispan.factories.ComponentRegistry.getFactory(ComponentRegistry.java:170)
	at org.infinispan.factories.AbstractComponentRegistry.getOrCreateComponent(AbstractComponentRegistry.java:270)
```

Perhaps you could give me some hint how to deal with it?